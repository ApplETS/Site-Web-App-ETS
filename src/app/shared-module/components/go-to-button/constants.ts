export const LOGO_FOLDER = 'assets/images/logos';
export const FACEBOOK_ICON = LOGO_FOLDER + '/facebook-logo.png';
export const TWITTER_ICON = LOGO_FOLDER + '/twitter-logo.png';
export const GITHUB_OCTOCAT_ICON = LOGO_FOLDER + '/github-white-octocat-logo.png';
export const GITHUB_ICON = LOGO_FOLDER + '/github-logo.png';

export const FACEBOOK_LABEL = 'Facebook logo';
export const TWITTER_LABEL = 'Twitter logo';
export const GITHUB_LABEL = 'Github logo';
