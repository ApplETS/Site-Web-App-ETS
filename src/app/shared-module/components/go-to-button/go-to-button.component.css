.container{
    display: flex;
    align-items: center;
    padding: 10px;
    border: 1px solid rgba(255, 255, 255, 0.534);
    border-radius: 5px;
    cursor:pointer;
    text-decoration-line: none;
    transition:background-color 300ms;
    color:white;
    width: fit-content;
}

.container:hover{
    background-color:white;
    color:#2C2E2F;
}

.logo-image{
    margin-right:10px;
    height:2rem;
    border-radius:3px;
}

@media (max-width: 992px) {
    .logo-image{
        height:1.5rem;
    }
}
