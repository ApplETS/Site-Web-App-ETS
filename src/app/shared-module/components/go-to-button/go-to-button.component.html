<div>
    <a class="container" href='{{link}}'>
        <img class="logo-image" src='{{icon}}' alt='{{iconAlt}}'>
        <span style="text-decoration-line:none">{{text}}</span>
    </a>
    <div class="spacer-horizontal"></div>
</div>
