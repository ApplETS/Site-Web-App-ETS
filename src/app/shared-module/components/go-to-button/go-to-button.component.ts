import { Component, Input, OnInit } from '@angular/core';
import * as constants from './constants';

@Component({
  selector: 'app-go-to-button',
  templateUrl: './go-to-button.component.html',
  styleUrls: ['./go-to-button.component.css']
})
export class GoToButtonComponent implements OnInit {

  @Input() link: string;

  @Input() text: string;

  /**
   * The icon is automatically choose if the link is from facebook, twitter or github
   */
  @Input() icon?: string;
  @Input() iconAlt?: string;

  ngOnInit() {
    if (!this.icon && this.link) {
      if (this.link.includes('facebook')) {
        this.icon = constants.FACEBOOK_ICON;
        this.iconAlt = constants.FACEBOOK_LABEL;
      } else if (this.link.includes('twitter')) {
        this.icon = constants.TWITTER_ICON;
        this.iconAlt = constants.TWITTER_LABEL;
      } else if (this.link.includes('github')) {
        this.iconAlt = constants.GITHUB_LABEL;
        if (this.text !== '') {
          this.icon = constants.GITHUB_OCTOCAT_ICON;
        } else {
          this.icon = constants.GITHUB_ICON;
        }
      }
    }
  }

}
