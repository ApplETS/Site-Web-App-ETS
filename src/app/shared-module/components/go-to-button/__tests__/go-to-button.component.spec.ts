import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { GoToButtonComponent } from '../go-to-button.component';

describe('GoToButtonComponent', () => {
  let component: GoToButtonComponent;
  let fixture: ComponentFixture<GoToButtonComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [GoToButtonComponent]
    })
      .compileComponents();

    fixture = TestBed.createComponent(GoToButtonComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  }));

  it('should create', () => {
    expect(fixture).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  describe('Auto icon and icon alt', () => {
    // TODO
  });
});
