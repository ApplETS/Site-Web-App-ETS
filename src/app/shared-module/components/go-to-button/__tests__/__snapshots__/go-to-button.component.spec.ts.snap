// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoToButtonComponent should match snapshot 1`] = `
<app-go-to-button
  iconAlt=""
>
  <a
    class="container"
    href=""
  >
    <img
      alt=""
      class="logo-image"
      src=""
    />
    <span
      style="text-decoration-line:none"
    >
      
    </span>
  </a>
</app-go-to-button>
`;
